### PR TITLE
TIP-1298: Adds domain information

### PR DESCRIPTION
**Pull request by BitOne**
**Description (for Contributor and Core Developer)**
Adds domain information to analytics.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Done
| Added legacy Behats               | -
| Added acceptance tests            | Done
| Added integration tests           | Done
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Platform\Bundle\AnalyticsBundle\Query;
```
Comment (ahocquard) [inline src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php]: `Query/Sql` to respect the port and adapter pattern

File: src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            SELECT DISTINCT(SUBSTRING_INDEX(email, '@', -1)) AS email_domain
+                FROM oro_user
+                ORDER by email_domain
+SQL;
+
+        $rows = $this->connection->fetchAll($sql);
+
+        foreach ($rows as $row) {
+            $results[] = $row['email_domain'];
+        }
```
Comment (ahocquard) [inline src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php]: `$this->connection->executeQuery($query)->fetchAll(\PDO::FETCH_COLUMN, 0)` avoid to do the foreach FYI

File: src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+        return implode(',', $results);
```
Comment (ahocquard) [inline src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php]: I think this is just for the format of the API response. I would return an array in the query.

File: src/Akeneo/Platform/Bundle/AnalyticsBundle/DataCollector/DBDataCollector.php [hunk 4]
```diff
@@ -138,6 +128,7 @@ public function collect()
             'avg_product_values_by_product'  => $this->productValueAverageMaxQuery->fetch()->getAverageVolume(),
             'avg_product_values_by_family'  => $this->productValuePerFamilyAverageMaxQuery->fetch()->getAverageVolume(),
             'max_product_values_by_family'  => $this->productValuePerFamilyAverageMaxQuery->fetch()->getMaxVolume(),
+            'email_domains' => $this->emailDomains->fetch()
```
Comment (ahocquard) [inline src/Akeneo/Platform/Bundle/AnalyticsBundle/DataCollector/DBDataCollector.php]: why is it a string and not an array?

File: src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php (hunk 3 above)
Comment (BitOne) [reply src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php]: This is much difficult to analyze after that: the content is sent as this directly.

File: src/Akeneo/Platform/Bundle/AnalyticsBundle/DataCollector/DBDataCollector.php (hunk 4 above)
Comment (BitOne) [reply src/Akeneo/Platform/Bundle/AnalyticsBundle/DataCollector/DBDataCollector.php]: This is much difficult to analyze after that: the content is sent as this directly.

File: src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php (hunk 1 above)
Comment (BitOne) [reply src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php]: Okay, I will fix it on master.

File: src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php (hunk 2 above)
Comment (BitOne) [reply src/Akeneo/Platform/Bundle/AnalyticsBundle/Query/EmailDomains.php]: Okay, I will fix it on master.

**Pull request merged**